src/Core/Models/Request/TokenRequest.cs

```diff
@@ -15,13 +15,14 @@ public class TokenRequest
         public string CodeVerifier { get; set; }
         public string RedirectUri { get; set; }
         public string Token { get; set; }
+        public string AuthRequestId { get; set; }
         public TwoFactorProviderType? Provider { get; set; }
         public bool? Remember { get; set; }
         public string CaptchaToken { get; set; }
         public DeviceRequest Device { get; set; }
 
         public TokenRequest(string[] credentials, string[] codes, TwoFactorProviderType? provider, string token,
-            bool? remember, string captchaToken, DeviceRequest device = null)
+            bool? remember, string captchaToken, string authRequestId = null, DeviceRequest device = null)
```

Comment:
Put the new parameter to the end, I see you had to add explicit parameter naming on several places because of this change. If you move it to the end, then all places stay as is except the ones where you need to pass the `authRequestId`. When you do that remove the explicit parameter naming of `device` when there's no need of it